soc/arm/nxp_s32/Kconfig

```diff
@@ -10,6 +10,36 @@ config SOC_FAMILY
 	string
 	default "nxp_s32"
 
+config NXP_S32_PM
```

Comment:
Why are we introducing a new `Kconfig`. Can we reuse `CONFIG_PM`

Reply:
option removed, please see https://github.com/zephyrproject-rtos/zephyr/pull/58332#discussion_r1252374694